packages/enzyme/src/ReactWrapper.js

```diff
@@ -802,6 +802,27 @@ class ReactWrapper {
     return this.props()[propName];
   }
 
+  /**
+   * Used to invoke a function prop.
+   * Will invoke an function prop and return its value.
+   *
+   * @param {String} propName
+   * @returns {Any}
+   */
+  invoke(propName) {
+    return this.single('invoke', () => {
+      const handler = this.prop(propName);
+      if (typeof handler !== 'function') {
+        throw new TypeError('ReactWrapper::invoke() expects a function prop name as its first argument');
+      }
+      return (...args) => {
+        const response = handler.apply(this, args);
```

Comment:
why should `this` be passed? if anything, it'd get `this.props`, but i think it'd be better to do:
```suggestion
        const response = handler(...args);
```